packages/vite-node/src/cli.ts

```diff
@@ -70,6 +71,7 @@ async function run(options: CliOptions = {}) {
     logLevel: 'error',
     configFile: options.config,
     root: options.root,
+    plugins: [RemoveShebangPlugin()],
```

Comment:
We directly confirm what code is run in [vitest/packages/vite-node/src/client.ts:181](https://github.com/vitest-dev/vitest/blob/main/packages/vite-node/src/client.ts#L181), so we don't need any plugins here. Also, this will only work for `vite-node`, but not for `vitest`

Reply:
> We directly edit what code is run in [vitest/packages/vite-node/src/client.ts:181](https://github.com/vitest-dev/vitest/blob/main/packages/vite-node/src/client.ts#L181)

The parse error occured when getting the variable `transformed` which would be embeded in `vm.runInThisContext`.
https://github.com/vitest-dev/vitest/blob/22f3c29610b415f78c8578365eb1a5aaeb0c8d59/packages/vite-node/src/client.ts#L132

 Inside `await this.options.fetchModule(id)`,  functions exported from vite are used, which can only be interfered with via plugin from `vite-node`.

> this will only work for vite-node, but not for vitest

Yes, I was thinking of the following use case with `vite-node` and had not considered about `vitest`.

```
// cli.ts
#!/usr/bin/env node
import("../your_module")
```

```shell
// users can run .ts code without thinking ts when developing cli tool.
$ npx vite-node cli.ts
```

Reply:
> Yes, I was thinking of the following use case with vite-node and had not considered about vitest.

Don't you want to also test that code 🤔 

> The parse error occured when getting the variable transformed which would be embeded in vm.runInThisContext.

Interesting, I though error happens when we call `vm.runInThisContext` 🤔 Maybe, this is a Vite bug, then? Does regular Vite files work?

Reply:
I found a solution ✨

`vite-node` uses the SSR mode of `vite` to transform. In doing so, the error occurred throw by the `acorn` parser.

https://github.com/vitejs/vite/blob/dddda1e05dd2184960afdcdf2960053acb30f5a9/packages/vite/src/node/ssr/ssrTransform.ts#L37

But, I found the [`allowHashBang`](https://github.com/acornjs/acorn/issues/629) option in `acron`. If it's enabled, 
we can treat shebang in `vite` and `vite-node` .

Also, `vm.runInthisContext` treates shebang as an Invalid token. You are right :)
```
SyntaxError: Invalid or unexpected token
    at new Script (node:vm:100:7)
    at createScript (node:vm:257:10)
    at Object.runInThisContext (node:vm:305:10)
    at ViteNodeRunner.directRequest (file:///Users/toueyoshiaki/Desktop/OSS/vitest/packages/vite-node/dist/client-900fb7db.js:144:19)
```

Therefore, need to remove shebang from the variable `transformed and both `vite-node` and `vitest` can run the code with shebang.

In conclusion, 
1: add allowHashBang in vite
2: change `transformed` => `transformed.replace(/^\#\!.*/, "")

should fix the bug and no plugins are required !!